custom_components/octopus_energy/api_client/__init__.py

```diff
@@ -657,12 +657,14 @@ async def async_get_smart_meter_consumption(self, device_id: str, period_from: d
         response_body = await self.__async_read_response__(live_consumption_response, url)
 
         if (response_body is not None and "data" in response_body and "smartMeterTelemetry" in response_body["data"] and response_body["data"]["smartMeterTelemetry"] is not None and len(response_body["data"]["smartMeterTelemetry"]) > 0):
-          return list(map(lambda mp: {
-            "consumption": float(mp["consumptionDelta"]) / 1000,
-            "demand": float(mp["demand"]) if "demand" in mp and mp["demand"] is not None else None,
-            "start": parse_datetime(mp["readAt"]),
-            "end": parse_datetime(mp["readAt"]) + timedelta(minutes=30)
-          }, response_body["data"]["smartMeterTelemetry"]))
+          return [
+            {
+              "consumption": float(mp["consumptionDelta"]) / 1000,
+              "demand": float(mp["demand"]) if "demand" in mp and mp["demand"] is not None else None,
+              "start": parse_datetime(mp["readAt"]),
+              "end": parse_datetime(mp["readAt"]) + timedelta(minutes=30)
+            } for mp in response_body["data"]["smartMeterTelemetry"] if mp.get("consumptionDelta")
```

Comment:
The issue with this approach is that if consumption is not defined, we'll fail to get/update the demand as well which might be confusing for users (as they'll expect it to update every minute).

It would be best to make `consumption` optional and then update all references of this data to account for consumption being optional (e.g. where we accumulate the consumption data, we should skip over it).